### ISSUE TITLE
@keyref attribute no longer processed to @href on <mathmlref>

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#2391]
So my small DITA topic uses the new mathmlref element:

    <equation-inline>
      <mathml>
        <mathmlref keyref="gigi"/>
      </mathml>
    </equation-inline>

and in the temporary files folder the topic should contain an extra @href on the "mathmlref", @href added by the "KeyrefPaser".
This does not happen because the method:

org.dita.dost.reader.GenListModuleReader.startElement(String, String, String, Attributes)

has an if clause which skips the "mathmlref" from processing because it considers it to be inside a foreign element (because the parent "mathml" element is an extension of foreign):

        if (foreignLevel > 0) {
            foreignLevel++;
            return;
        }

So the "GenListModuleReader" reports my topic has no keyref and it will no longer be processed to add an @href to it.

**Comment by raducoravu** [comment]
The specs says that the foreign element can contain a mixture of DITA and non-DITA elements:
https://www.oxygenxml.com/dita/1.3/specs/non-normative/foreigngeneralization.html

**Comment by raducoravu** [comment]
Any input on this?

**Comment by jelovirt** [comment]
One of the problems with foreign is that the spec doesn't say how to recognize DITA elements inside foreign. One option is to check that it has `@class` attribute that has a valid DITA class value, even though non-DITA element might pass this test.

**Comment by raducoravu** [comment]
I agree, what you suggest is the only way to go. Check the @class attribute value, if it does not match the DITA pattern ignore the element and if it does consider it DITA.

**Comment by raducoravu** [comment]
Or simply add specific cases for the mathml and foreign elements. Something like:

    else if ((TOPIC_FOREIGN.matches(classValue) || TOPIC_UNKNOWN.matches(classValue))
        		//OXYGEN PATCH for EXM-37092 and EXM-36738
        		&& (! classValue.contains(" svg-d/") && ! classValue.contains(" mathml-d/")))

Also in the same class the method org.dita.dost.reader.GenListModuleReader.getFormat(Attributes) should be changed like:

        if (TOPIC_IMAGE.matches(attrClass) 
        		|| attrClass.contains(" svg-d/svgref ") || attrClass.contains(" mathml-d/mathmlref ")) {
            return ATTR_FORMAT_VALUE_IMAGE;

**Comment by raducoravu** [comment]
This is important for DITA 1.3 compatibility.
